service/storage_service.cc

```diff
@@ -791,20 +791,15 @@ future<> storage_service::sstable_cleanup_fiber(raft::server& server, sharded<se
 
                 rtlogger.info("start cleanup");
 
-                auto keyspaces = _db.local().get_all_keyspaces();
-
-                tasks.reserve(keyspaces.size());
-
                 co_await _db.invoke_on_all([&] (replica::database& db) {
                     return db.flush_all_tables();
                 });
-                co_await coroutine::parallel_for_each(keyspaces.begin(), keyspaces.end(), [this, &tasks, &do_cleanup_ks] (const sstring& ks_name) -> future<> {
+                // Skip tablets tables since they do their own cleanup and system tables
+                // since they are local and not affected by range movements.
+                auto ks_erms = _db.local().get_non_local_strategy_keyspaces_erms();
```

Comment:
Should probably get the erms before flushing, so that we don't cleanup new keyspaces which were not flushed.

Reply:
agreed. but i am curious what would be the problem if we cleanup keyspaces which are not flushed. the reason why i moved it around was that i wanted to move `ks_erms` closer to where it is used.

anyway, after a second thought, it might be better off preserving existing behavior, and only do what this change claims to do. just to be on the safe side.

Reply:
I don't know what would happen, but if order didn't matter then we wouldn't need to flush.

Reply:
thanks!